app/graphql/resolvers/freaks_resolver.rb

```diff
@@ -2,11 +2,17 @@
 
 module Resolvers
   class FreaksResolver < Resolvers::Base
+    argument :filter, Types::FreakFilterType, required: false
+
     type Types::FreakType.connection_type, null: false
 
     # :reek:UtilityFunction
-    def resolve
-      Freak.all
+    def resolve(filter: nil)
```

Comment:
As a rule of thumb, avoid working with `nil`/`null` if possible. You can get in all sorts of issues - fo example an empty array `[]` is better than `nil`, since calling `nil.first` will raise an error , while `[].first` won't .

Here's some java docs on this concept: http://www.javapractices.com/topic/TopicAction.do?Id=134 (check the `See also` section).

An empty hash would be closer to the data type you're working with.

Reply:
I replaced nil with an empty hash. Thanks for the advice with nil usage :slightly_smiling_face: